src/Application/Migrations/15_function_ticket_state.sql

```diff
@@ -39,21 +48,30 @@ BEGIN
 	SELECT
 		pts.ticket_state, pts.service_executable
 	FROM
-		tbl_ticket_state ts1
+		tbl_ticket t
+	JOIN
+		tbl_ticket_state ts_this ON ts_this.ticket_type = t.ticket_type AND ts_this.ticket_state = COALESCE(param_ticket_state, t.ticket_state)
 	JOIN
-		tbl_project_ticket_state pts ON pts.ticket_type = ts1.ticket_type AND pts.ticket_state = ts1.ticket_state
+		tbl_project_ticket_state pts ON pts.project_id = t.project_id AND pts.ticket_type = t.ticket_type
 	JOIN
-		tbl_ticket_state ts2 ON ts1.ticket_type = ts2.ticket_type AND ts1.sort < ts2.sort
+		tbl_ticket_state ts_other ON ts_other.ticket_type = pts.ticket_type AND ts_other.ticket_state = pts.ticket_state
 	WHERE
-		pts.project_id = param_project_id AND
-		ts2.ticket_type = param_ticket_type AND
-		ts2.ticket_state = param_ticket_state
-  ORDER BY
-    ts1.sort DESC
+		t.id = param_ticket_id AND
+		ts_other.sort < ts_this.sort AND
+		(pts.skip_on_dependent = FALSE OR
+		( /* is master encoding ticket */
+			SELECT ep.depends_on
+			FROM tbl_ticket t
+			JOIN tbl_encoding_profile_version epv ON epv.id = t.encoding_profile_version_id
+			JOIN tbl_encoding_profile ep ON ep.id = epv.encoding_profile_id
+			WHERE t.id = param_ticket_id
+		) IS NULL )
```

Comment:
Second time the exact same subquery is used. Would it be worth it to have is `is_master_ticket(ticket_id)`?